### PR TITLE
Add django-tos contrib app

### PR DESCRIPTION
**Pull request by blag**
This PR adds an integration with [django-tos](/revsys/django-tos/).

It is necessary when using django-tos's middleware option (2). When using django-tos' middleware, since guest users have full user accounts they are required to agree to the site's terms of service before they can do anything else.

I have refactored django-tos a bit in revsys/django-tos#65 specifically to make this custom middleware easier to write. That PR allows us to minimize the amount of code necessary in this middleware and any changes to upstream django-tos should not interfere with this middleware.

### CODE REVIEW

**Comment by julianwachholz** [comment]
Hi @blag thank you for this contribution! I think this could be added but if I understand it correctly it depends on the linked PR being merged into `django-tos` first?

**Comment by blag** [comment]
Sorry I wasn't clear about that - yes, it needs the linked PR to be merged first. That maintainer is pretty active though, so it hopefully shouldn't take too much longer.

File: docs/contrib.rst [hunk 1]
```diff
@@ -69,3 +69,45 @@ As an example, this handler will save the avatar URL.
...
+This submodule is not enabled by default. To enable it, add it to your
+``INSTALLED_APPS``.
+
+.. code:: python
+
+  INSTALLED_APPS = [
+      "tos",
+      # ...
+      "guest_user",
+      "guest_user.contrib.tos",
```
Comment (julianwachholz) [inline docs/contrib.rst]: Do you prefer adding the contrib app here even if it's not required?

Comment (blag) [reply docs/contrib.rst]: Good catch. I simplified the setup instructions.

The full path to the middleware does now require `__init__.py` files in a lot more places - I hope that's acceptable.

**Comment by julianwachholz** [comment]
Thanks for your contribution and continued support of this project @blag :)

**Comment by blag** [comment]
Thank you for maintaining it for me and the Django community. Happy New Year! 😄

**Pull request merged**